File: tests/Generator/resource/typescript/complex/ScalarPropertyType.ts
import {BooleanPropertyType} from "./BooleanPropertyType";
import {IntegerPropertyType} from "./IntegerPropertyType";
import {NumberPropertyType} from "./NumberPropertyType";
import {StringPropertyType} from "./StringPropertyType";
import {PropertyType} from "./PropertyType";

/**
 * Base scalar property type
 */
export interface ScalarPropertyType extends PropertyType {
    type?: string
}

File: tests/Generator/resource/rust/complex/scalar_property_type.rs
use serde::{Serialize, Deserialize};
use boolean_property_type::BooleanPropertyType;
use integer_property_type::IntegerPropertyType;
use number_property_type::NumberPropertyType;
use string_property_type::StringPropertyType;
use property_type::PropertyType;

// Base scalar property type
#[derive(Serialize, Deserialize)]
pub struct ScalarPropertyType {
    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "nullable")]
    nullable: Option<bool>,

    #[serde(rename = "type")]
    _type: Option<String>,

}

